src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHostUserInterface.cs

```diff
@@ -1468,13 +1469,16 @@ private string ReadLineFromConsole(bool endOnTab, string initialContent, bool ca
             _rawui.ClearKeyCache();
             uint keyState = 0;
             string s = string.Empty;
+            Span<char> inputBuffer = stackalloc char[MaxInputLineLength + 1];
+            initialContent.AsSpan().CopyTo(inputBuffer);
```

Comment:
This should be moved into the `do..while` loop, and we should do this only if `initialContent.Length > 0`.

Reply:
Could you please clarify? I raised it one level up from ConsoleControl.ReadConsole(). Every call the buffer was allocated (with initialContent) what was inevitable. Now we can exclude the extra operation because the buffer is modifed only by users and is not modifed in the method. Also do-while is not used on Windows - all branches have `break` operator.

> initialContent.Length > 0

It seems all the methods have AggressiveInlining - will look asm.

Reply:
Yes, I misunderstood the behavior of `CONSOLE_READCONSOLE_CONTROL.nInitialChars`. No need to move this line.